internal/manifest/manifest.go

```diff
@@ -21,9 +21,15 @@ type Manifest struct {
 }
 
 type Component struct {
-	Name string `hcl:"name,label"`
-	Type string `hcl:"type,label"`
-	Spec string `hcl:"spec"` // TODO: Custom unmarshalling to allow convenient json representation.
+	Name      string         `hcl:"name,label"`
+	Type      string         `hcl:"type,label"`
+	Spec      string         `hcl:"spec"` // TODO: Custom unmarshalling to allow convenient json representation.
+	DependsOn []ComponentRef `hcl:"depends_on"`
+}
+
+type ComponentRef struct {
+	Name string `hcl:"name"`
+	Type string `hcl:"type"`
```

Comment:
As far as I can tell, you never read out this `Type` field, but maybe I missed it somewhere? Since there is only one namespace for all types, I don't think ComponentRef should have a Type field at all. Is there a reason you added it that I'm missing?